modules/custom/d_content/modules/d_content_init/src/Services/ConfigUpdate.php

```diff
@@ -41,65 +42,138 @@ public function __construct(ConfigFactory $configFactory, ModuleHandler $moduleH
-        foreach ($files as $file) {
-          if (preg_match($regex, $file) === 1) {
-            $configs[] = substr($file, 0, -4);
-          }
-        }
+  public function getConfigsFilesName($moduleName, $path, $regex) {
+    $configsFileNames = [];
+    $filesNames = scandir($this->getConfigsPath($moduleName, $path));
+    foreach ($filesNames as $fileName) {
+      if (preg_match($regex, $fileName) === 1) {
+        $configsFileNames[] = substr($fileName, 0, -4);
       }
     }
-    return $configs;
+    return $configsFileNames;
   }
 
   /**
-   * Set default theme for configs.
+   * Create configurations for block for active theme.
    *
    * @param array $configs
-   *   Array of configuration file names.
+   *   Array of configs.
+   * @param string $path
+   *   The path to the directory with the configuration for the module.
    */
-  public function setTheme(array $configs) {
+  public function createBlocksConfigs(array $configs, $path) {
     $theme = $this->configFactory->get('system.theme')->get('default');
-    foreach ($configs as $config) {
-      $configToUpdate = $this->configFactory->getEditable($config);
-      if (!$configToUpdate->isNew()) {
-        $configToUpdate->set('theme', $theme)
-          ->set('dependencies.theme', $theme)
+    $storage = new FileStorage($path);
+    foreach ($configs as $themeConfigName) {
+      $subthemeConfigName = $this->getConfigName($themeConfigName, $theme);
+      $newConfig = $this->configFactory->getEditable($subthemeConfigName);
+      // Check if the configuration is new.
+      if ($newConfig->isNew()) {
+        $newConfig->setData($storage->read($themeConfigName));
+        // TODO: Add correct id for block config.
+        $newConfig->set('theme', $theme)
+          ->set('dependencies.theme', [$theme])
+          ->set('id', $this->getConfigId($subthemeConfigName))
           ->save();
       }
     }
   }
 
   /**
-   * Changes the value of the theme option to the default theme.
+   * Get config name parts.
+   *
+   * @param string $config
+   *   Configuration file names.
+   *
+   * @return array
+   *   Config name parts exploded by dots.
+   */
+  public function getConfigNameParts($config) {
+    return explode('.', $config);
+
+  }
+
+  /**
+   * Get config name for active theme.
+   *
+   * @param string $config
+   *   Configuration file names.
+   * @param string $theme
+   *   Active theme name.
+   *
+   * @return string
+   *   Config name for theme.
+   */
+  public function getConfigName($config, $theme) {
+    $parts = $this->getConfigNameParts($config);
+    // Add active theme name to config.
+    $parts[2] = $theme . '_' . $parts[2];
+    return implode('.', $parts);
+  }
+
+  /**
+   * Get config id from config name for active theme.
+   *
+   * @param string $config
+   *   Config name.
+   *
+   * @return string
+   *   Config id.
+   */
+  public function getConfigId($config) {
+    $parts = $this->getConfigNameParts($config);
```

Comment:
Na pewno zawsze dwa pierwsze komponenty będą do ucięcia? Może moglibyśmy brać to ID z samego pliku yml?

Reply:
Sprawdziłem jak wyglądają nazwy plików konfiguracji w innych modułach i zawsze dwie pierwsze częci nazwy łączone kropką określają typ konfiguracji

Reply:
Na razie masz zawsze, ale jest dużo configów typu:
`modulename.settings.yml`
z cora też masz np.
`core.extensions.yml`
więc jak utniesz dwa, to nic Ci nie zostaje
Raczej bym szedł w kierunku tego co @grzegorz-pietrzak-droptica sugeruje.